app/bundles/PageBundle/Controller/PublicController.php

```diff
@@ -545,6 +546,11 @@ public function redirectAction(
                 $leadArray = ($lead) ? $primaryCompanyHelper->getProfileFieldsWithPrimaryCompany($lead) : [];
 
                 $url = TokenHelper::findLeadTokens($url, $leadArray, true);
+
+                // Dispatch URL token replace event to allow modifications
+                $urlEvent = new UrlTokenReplaceEvent($url, $lead, null);
+                $this->dispatcher->dispatch($urlEvent, PageEvents::ON_URL_TOKEN_REPLACE);
```

Comment:
```suggestion
                $this->dispatcher->dispatch($urlEvent);
```
In newer Symfony versions you don't need a special name for the event. The Event class itself do the trick.